### PR TITLE
ks: init at 0.0.4

### PR DESCRIPTION
**Pull request by ivankovnatsky**
## Description of changes

Adds new OS specific package for Darwin only. Command-line secrets manager powered by macOS keychains. 
Basically a wrapper around builtin Darwin `security` CLI command.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by kirillrdy** [comment]
@infinisil interesting edge case for by-name check

**Comment by kirillrdy** [comment]
@ivankovnatsky i think your original approach was pretty valid,  maybe wait for @infinisil to comment

File: pkgs/os-specific/darwin/ks/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    rev = "${version}";
+    hash = "sha256-jGo0u0wiwOc2n8x0rvDIg1suu6vJQ5UCfslYD5vUlyI=";
+  };
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ${pname} $out/bin/
+  '';
+
+  meta = with lib; {
```
Comment (kirillrdy) [inline pkgs/os-specific/darwin/ks/default.nix:22]: consider adding `mainProgram` https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

File: pkgs/os-specific/darwin/ks/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+, fetchFromGitHub
+}:
+
+stdenv.mkDerivation rec {
+  pname = "ks";
+  version = "0.4.0";
+
+  src = fetchFromGitHub {
+    owner = "loteoo";
+    repo = pname;
```
Comment (kirillrdy) [inline pkgs/os-specific/darwin/ks/default.nix:12]: usage of pname considered anti pattern https://github.com/NixOS/nixpkgs/issues/277994

File: pkgs/os-specific/darwin/ks/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+}:
+
+stdenv.mkDerivation rec {
```
Comment (RaghavSood) [inline pkgs/os-specific/darwin/ks/default.nix:6]: For a `mkDerivation`, it's preferable to use `finalAttrs` instead of `rec`

**Comment by kirillrdy** [comment]
@ivankovnatsky can you swash commits

**Comment by ivankovnatsky** [comment]
> @ivankovnatsky can you swash commits

Sure, thanks.

**Review by RaghavSood — APPROVED** [review]

**Comment by RaghavSood** [comment]
Result of `nixpkgs-review pr 312401` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by RaghavSood** [comment]
Result of `nixpkgs-review pr 312401` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/by-name/ks/ks/package.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    cp ${finalAttrs.pname} $out/bin/
+  '';
+
+  meta = {
+    mainProgram = "ks";
+    homepage = "https://github.com/loteoo/ks";
+    description = "Command-line secrets manager powered by macOS keychains";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ ivankovnatsky ];
+    platform = lib.platforms.darwin;
```
Comment (kirillrdy) [inline pkgs/by-name/ks/ks/package.nix:28]: CI meta check is failing because platform is not a valid meta attribute it should be `platforms`

Comment (ivankovnatsky) [reply pkgs/by-name/ks/ks/package.nix]: Updated, can you point please me how to check that, so I could do it next time first.

Comment (RaghavSood) [reply pkgs/by-name/ks/ks/package.nix]: You can check the ofborg log for the failing job by following the link from the ❌ job on your previous [commit](https://github.com/NixOS/nixpkgs/commit/4b56de9450627d3daf25343fe7724d0ee36e1eb9): https://gist.github.com/GrahamcOfBorg/42df16c9ec93f97612757c02d941c9c9

Comment (kirillrdy) [reply pkgs/by-name/ks/ks/package.nix]: image

it already started re-eval due to new changes being pushed, but one of checks is metadata check, and it was failing

**Pull request merged**